src/Transactions/TransactionDetails/TransactionInformation/SourceValue/useSourceValue.js

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from 'react-query';
+
+import { useOkapiKy } from '@folio/stripes/core';
+import {
+  LINES_API,
+  INVOICES_API,
+} from '@folio/stripes-acq-components';
+
+import { TRANSACTION_SOURCE } from '../../../constants';
+
+export const useSourceValue = (transaction) => {
+  const ky = useOkapiKy();
+  const isInvoiceSource = transaction.source === TRANSACTION_SOURCE.invoice;
+
+  const { isLoading: isInvoiceLoading, data: invoice } = useQuery(
+    transaction.id,
+    () => ky(`${INVOICES_API}/${transaction.sourceInvoiceId}`).json(),
+    { enabled: isInvoiceSource },
+  );
+
+  const { isLoading: isPoLineLoading, data: poLine } = useQuery(
+    transaction.id,
+    () => ky(`${LINES_API}/${transaction.encumbrance?.sourcePoLineId}`).json(),
+    { enabled: transaction.source === TRANSACTION_SOURCE.poLine },
+  );
```

Comment:
two queries with the same key?

Reply:
Ah, I guess it's because only one of the queries is run (enabled) depending on transaction.source

Reply:
Maybe something like that?
```js
const config = {
  [TRANSACTION_SOURCE.invoice]: {
    path: `${INVOICES_API}/${transaction.sourceInvoiceId}`,
    field: 'folioInvoiceNo',
  },
  [TRANSACTION_SOURCE.poLine]: {
    path: `${LINES_API}/${transaction.encumbrance?.sourcePoLineId}`,
    field: 'poLineNumber',
  }
}

const path = config[transaction.source]?.path;

const { isLoading, data } = useQuery(path, () => ky(path).json(), { enabled: path });

const sourceValue = data?.[config[transaction.source]?.field]

return ({ isLoading, sourceValue });
```

I think we have one react-query cache scope over the whole folio, so making the key just a `transaction.id` can mess things.

Reply:
I agree that code is better to rewrite to something extendable, but I would make it easier -
```
const queryFnsMap = {
  [TRANSACTION_SOURCE.invoice]: (ky, transaction) => {
    const {folioInvoiceNo} = await ky(`${INVOICES_API}/${transaction.sourceInvoiceId}`).json();
    return folioInvoiceNo;
  },
  ...
};

const useSourceValue = (transaction) => {
  let result = {isLoading: fasle, sourceValue: null};
  const queryFn = queryFnsMap[transaction.source];
  if (queryFn) {
    result = useQuery(['finance', 'transaction-source-value', transaction.id], queryFn);
  }
  return result;
};
```

Reply:
thanks, guys for sharing ideas.